server/routerlicious/packages/services-core/src/checkpointService.ts

```diff
@@ -0,0 +1,177 @@
+        const checkpointFilter = {
+            documentId,
+            tenantId
+        };
+
+        await (isLocal && this.localCheckpointEnabled ? this.checkpointRepository.removeServiceCheckpoint(documentId, tenantId)
+            .catch((error) => {
+                Lumberjack.error(`Error removing checkpoint from local databse.`, lumberProperties, error);
+            }) : this.documentRepository.updateOne(
+                checkpointFilter, {
+                    [service]: ""
+                }
+            ));
+    }
+
+    async restoreFromCheckpoint(documentId: string, tenantId: string, service: string, document: IDocument) {
+        let checkpoint;
+        let lastCheckpoint: IDeliState | IScribe;
+        let isLocalCheckpoint = false;
+        const restoreFromCheckpointMetric = Lumberjack.newLumberMetric(LumberEventName.RestoreFromCheckpoint);
+        let checkpointSource = "defaultGlobalCollection";
+
+        try {
+            if (!this.localCheckpointEnabled || !this.checkpointRepository){
+                // If we cannot checkpoint locally, use document
+                lastCheckpoint = JSON.parse(document[service]);
+            } else {
+                // Search checkpoints collection for checkpoint
+                checkpoint = await this.checkpointRepository.getCheckpoint(documentId, tenantId)
+                .catch((error) => {
+                    Lumberjack.error(`Error retrieving local checkpoint`, getLumberBaseProperties(documentId, tenantId));
+                    checkpointSource = "notFoundInLocalCollection";
+                });
+
+                if(checkpoint?.deli) {
+                    lastCheckpoint = JSON.parse(checkpoint[service]);
+                    checkpointSource = "foundInLocalCollection";
+                    isLocalCheckpoint = true;
+                } else {
+                    // If checkpoint does not exist, use document
+                    checkpointSource = "notFoundInLocalCollection";
+                    lastCheckpoint = JSON.parse(document[service]);
+                }
+
+            }
+            restoreFromCheckpointMetric.setProperties({
+                [BaseTelemetryProperties.tenantId]: tenantId,
+                [BaseTelemetryProperties.documentId]: documentId,
+                service,
+                checkpointSource,
+                retrievedFromLocalDatabase: isLocalCheckpoint,
+            });
+        } catch (error){
+            Lumberjack.error(`Error finding the last checkpoint.`, getLumberBaseProperties(documentId, tenantId), error)
+        }
+
+        if (lastCheckpoint) {
+            restoreFromCheckpointMetric.success(`Restored checkpoint from database.`);
+            return lastCheckpoint;
+        } else {
+            restoreFromCheckpointMetric.error(`Error restoring checkpoint from database. Last checkpoint not found.`);
+            return;
+        }
+
+
+    }
+
+    async getLatestCheckpoint(
```

Comment:
Could you apply this method to : https://github.com/microsoft/FluidFramework/blob/main/server/routerlicious/packages/services-utils/src/deleteSummarizedOps.ts#L44